langchain/schema.py

```diff
@@ -34,15 +35,22 @@ def get_buffer_string(
             role = ai_prefix
         elif isinstance(m, SystemMessage):
             role = "System"
+        elif isinstance(m, FunctionMessage):
+            role = "Function"
         elif isinstance(m, ChatMessage):
             role = m.role
         else:
             raise ValueError(f"Got unsupported message type: {m}")
-        string_messages.append(f"{role}: {m.content}")
+        message = f"{role}: {m.content}"
+        if isinstance(m, AIMessage) and "function_call" in m.additional_kwargs:
+            message += f"{m.additional_kwargs['function_call']}"
+        string_messages.append(message)
+
     return "\n".join(string_messages)
 
 
-class AgentAction(NamedTuple):
+@dataclass
+class AgentAction:
```

Comment:
Removal of the `NamedTuple` base class relationship (and making it a `@dataclass` instead) breaks the ["Accessing Intermediate Steps" How-To](https://python.langchain.com/docs/modules/agents/how_to/intermediate_steps), as AgentAction objects now are no longer JSON serializable.
<img width="618" alt="image" src="https://github.com/hwchase17/langchain/assets/569413/b51406fc-921a-4449-b55b-d117b7361f6f">

@hwchase17 Any thoughts on how to best address this?